src/run/troubleshoot/analysis/ProxiesConfiguredCorrectlyAnalysis.ts

```diff
@@ -0,0 +1,97 @@
+import * as fs from "fs";
+import * as os from "os";
+import * as path from "path";
+import { AnalysisIssue, TroubleshootAnalysisTaskContext } from "./AnalysisTask.js";
+import AnalysisOutcome from "./AnalysisOutcome.js";
+
+type DockerSettingsInterface = {
+    OverrideProxyHTTP: string;
+    OverrideProxyHTTPS: string;
+    ProxyHttpMode: string
+    ProxyHTTPMode: string
+}
+
+const ANALYSIS_HEADLINE = "Check user's device proxy configuration";
+const DOCKER_SETTINGS_FILE_SUGGESTIONS = [
+    "Check file path to docker 'settings-store.json'"
+];
+const DOCKER_PROXY_CONFIGURATION_SUGGESTIONS = [
+    "Run 'chproxyon' to set ProxyHttpMode for docker",
+    "'ProxyHttpMode' property in docker settings-store.json file must be set to 'manual'",
+    "'ProxyHTTPMode' property in docker settings-store.json file must be set to 'manual'",
+    "Run echo $HTTPS_PROXY to check web proxy value on device",
+    "'OverrideProxyHTTP' property in docker settings-store.json file must correctly match 'HTTPS_PROXY' value",
+    "'OverrideProxyHTTPS' property in docker settings-store.json file must correctly match 'HTTPS_PROXY' value"
+];
+
+/**
+ * An analysis task which checks whether the proxy configuration for the user device are configured correctly.
+ */
+
+export default class ProxiesConfiguredCorrectlyAnalysis {
+
+    async analyse (): Promise<AnalysisOutcome> {
+        const dockerIssues = this.checkDockerProxiesConfig();
+
+        const issues: AnalysisIssue[] = dockerIssues ? [dockerIssues] : [];
+
+        return this.createOutcomeFrom(issues);
```

Comment:
Are you actually checking the VPN is activated here? This is the task the docker stuff is nice to have. The user could have correct Docker settings and not be on the VPN. Slightly different to being on the VPN. - Perhaps a separate Analysis task?

Reply:
I am happy for the Docker config test to be separated out.

Reply:
I noticed the init hook get called and this line executed on every analysis command run which checks the vpn before line 34 : 
` if (isWebProxyHostSet()) {
        if (!isOnVpn()) {
            this.warn("Not on VPN. Some containers may not build properly.");
        }
    } else {
        this.warn("CH_PROXY_HOST env not set. Some containers may not build properly.");
    }`
  
The only thing is it doesnt return the error/warning in analysis output format.

Reply:
I think best to be overly verbose than assume people will action something which is no different to running any of the commands.

Reply:
I think the VPN check should be covered by analysis - can be separate PR if preferred but I think we should assume that if people have been ignoring the warning previously they will have ignored it now too. Also, the console will be busy with other things so will not be the clearest.

Reply:
I see your point. I will add it up